ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UpdateLiteralConstantFix.fs

```diff
@@ -0,0 +1,84 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open FSharp.Compiler.Symbols
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+
+type UpdateLiteralConstantFix(error: LiteralConstantValuesDifferError) =
+    inherit FSharpQuickFixBase()
+    let errorRefPat = error.Pat.As<IReferencePat>()
+    
+    let tryFindDeclarationFromSignature () =
+        let containingTypeDecl = errorRefPat.GetContainingTypeDeclaration()
+        let decls = containingTypeDecl.DeclaredElement.GetDeclarations()
+        decls |> Seq.tryFind (fun d -> d.GetSourceFile().IsFSharpSignatureFile)
+        
+    let tryFindSigBindingSignature sigMembers =
+        let p = errorRefPat.Binding.HeadPattern.As<IFSharpPattern>()
+        if Seq.length p.Declarations = 1 then
+            let implDec = Seq.head p.Declarations
+            let declName = implDec.DeclaredName
+            sigMembers
+            |>  Seq.tryPick(fun m ->
+                let bindingSignature = m.As<IBindingSignature>()
+                match bindingSignature with
+                | null -> None
+                | _ ->
+                    match bindingSignature.HeadPattern with
+                    | :? IReferencePat as sigRefPat when
+                        declName = sigRefPat.DeclaredName -> Some bindingSignature
+                    | _ -> None
+                )
+        else
+            None
+
+    let mutable sigRefPat = null
+
+    let rec isImplExprValidInSig (implExpression: IFSharpExpression) =
+        match implExpression with
+        | :? IReferenceExpr as refExpr ->
+            refExpr.Reference.ResolveWithFcs(sigRefPat, System.String.Empty, false, refExpr.IsQualified)
+            |> Option.isSome
+        | :? IBinaryAppExpr as binExpr ->
+            isImplExprValidInSig binExpr.LeftArgument && isImplExprValidInSig binExpr.RightArgument
+        | _ -> true
```

Comment:
We should probably be more defensive here. It's probably better to allow literal expressions and disallow the most of the other cases. What I mean is if there's some other compound syntax that is allowed in literal bindings, then we may skip the check if names are still resolved in the signature. Could you check if any other syntax is allowed?

Reply:
While still being a `ILiteralExpr`, there are also measures.
As we can't use `ResolveWithFcs` to check if the measure types are valid in the signature, I think we should not support them.

Reply:
> While still being a ILiteralExpr, there are also measures.

That's a really good observation! It seems our grammar lacks them for some reason. We should fix it, and the check will become easy to do. 🙂